src/jenkins.py

```diff
@@ -607,6 +607,23 @@ def _get_plugin_name(plugin_info: str) -> str:
     return match.group(1)
 
 
+def _wait_plugins_install(container: ops.Container, timeout: int = 60 * 5) -> None:
+    """Wait until all plugins are installed.
+
+    This function checks for any .tmp files in the plugins directory which indicates that a user
+    might be installing plugins through the UI.
+
+    Args:
+        container: The Jenkins workload container.
+        timeout: Timeout in seconds to wait for plugins to be installed.
+    """
+    _wait_for(
+        lambda: not container.list_files(path="/var/lib/jenkins/plugins/", pattern="*.tmp"),
```

Comment:
What happens if list_files fails?

Reply:
It should raise a PathError which should make the charm fall into an ErroredStatus. The charm should not be able to trigger this before the installation of Jenkins so if this path is missing, it should fall into an ErroredStatus :D

Reply:
I'm not familiar if this situation might easily happens or not but do you think is worthy creating a unit test for that case?

Reply:
WIll create a followup PR since it's break right now, the integration test has been added for a bigger scope for now :)